projects/plugins/jetpack/extensions/blocks/map/utils/get-map-provider.js

```diff
@@ -8,13 +8,18 @@ function getCookie( name ) {
 	}
 }
 
-const getMapProvider = () => {
+const getMapProvider = props => {
+	const mapboxStyles = [ 'black_and_white', 'terrain' ];
+	if ( props?.mapStyle && mapboxStyles.includes( props.mapStyle ) ) {
```

Comment:
I think both will work for this use case.

`mapboxStyles.includes( 'satellite' )` will return false
`mapboxStyles.includes( 'satellite' )` will return false

However:

```js
const mapboxStyles = [ 'black_and_white', 'terrain' ];
const mapStyle = "black_and_white_and_blue_this_shouldnt_be_true"; 
mapboxStyles.some( mapboxStyle => mapStyle.includes(mapboxStyle) );
````
Also returns `true` because the string contains `black_and_white`.

When used on a string (`props.mapStyle`) .includes() checks if the string contains a substring. On an array it checks if an element is in the array.

Reply:
Correct. I was just concerned that the `mapboxStyle` string can contain more than just that one class.

Reply:
Please disregard my above concern. I misunderstood what the `getActiveStyleName` function was doing [here](https://github.com/Automattic/jetpack/pull/29382/files#diff-7e61fb5ddc88a0cff10c780deef2738a28d45aff18b1a94894c95ed6b3188435R170-R174).